qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -279,6 +285,173 @@ def cancel(self) -> bool:
+
+        if update_successful:
+            self._name = updated_name  # Cache updated name if successful.
+
+        return update_successful
+
+    def modify_tags(self, tags: List[str], overwrite: bool = False) -> bool:
+        """Modify the tags associated with this job.
+
+        Note:
+            If a job was submitted through an
+            :class:`IBMQJobManager<qiskit.providers.ibmq.managed.IBMQJobManager>`
+            instance, then the job will contain a tag with the prefix ``ibmq_jobset_``.
+            These tags are necessary to retrieve a job within a
+            :class:`ManagedJobSet<qiskit.providers.ibmq.managed.ManagedJobSet>`.
+            As a result, tags that contain this prefix will not be overwritten
+            or changed.
+
+        An example of the behavior when using ``overwrite=False``::
+
+            current_job_tags = job.tags()  # Tags are: ['tag1', 'tag2']
+            job.modify_tags(['tag3'], overwrite=False)
+            new_job_tags = job.tags()  # Tags are: ['tag1', 'tag2', 'tag3']
+
+        An example of the behavior when using ``overwrite=True``::
+
+            current_job_tags = job.tags()  # Tags are: ['tag1', 'tag2']
+            job.modify_tags(['tag3'], overwrite=True)
+            new_job_tags = job.tags()  # Tags are: ['tag3']
+
+        Args:
+            tags: The new tags to add or `overwrite` for this job.
+            overwrite: If ``False`` the specified `tags` are added to the
+                job's already existing tags. If ``True``, the `tags` overwrite
+                and replace the tags currently associated with this job.
+
+        Returns:
+            ``True`` if modifying the job tags was successful, else ``False``.
+
+        Raises:
+            IBMQJobInvalidStateError: If the input job tags are invalid.
+            IBMQJobApiError: If an unexpected error occurred when communicating
+                with the server.
+        """
+        validate_job_tags(tags, IBMQJobInvalidStateError)
+
+        tags_to_update = set(tags)
+        if overwrite:
+            # Add the tags associated with a job set.
+            tags_to_update.update(
+                filter(lambda tag: re.match(RE_JOB_SET_ID_LONG, tag), self._tags))
+        else:
+            tags_to_update.update(self._tags)
+
+        with api_to_job_error():
+            response = self._api.job_update_attribute(
+                job_id=self.job_id(), attr_name='tags', attr_value=list(tags_to_update))
+
+        # Get the tags from the response and check if the update was successful.
+        updated_tags = response.get('tags', None)
+        updated_tags_set = set(updated_tags) if updated_tags else None
+        update_successful = (tags_to_update == updated_tags_set)
+
+        if update_successful:
+            self._tags = updated_tags  # Cache updated tags if successful.
+
+        return update_successful
+
+    def remove_tags(self, tags: List[str]) -> bool:
```

Comment:
I think you can consolidate this method with `modify_tags`, something like 
```
def update_tags(self, replacement_tags=None, additional_tags=None, remove_tags=None)
```

It's one fewer method for the users to remember, also easier to update a specific tag (e.g. changing "priority-high" to "priority-low").